docs/analyzer/adding_recognizers.md

```diff
@@ -0,0 +1,247 @@
+# Supporting new types of PII entities
+
+Presidio can be extended to support new types of PII entities, and to support additional languages.
+
+## Table of contents
+
+- [Introduction to recognizer development](#introduction-to-recognizer-development)
+- [Types of recognizer classes in Presidio](#types-of-recognizer-classes-in-presidio)
+- [Extending the analyzer for additional PII entities](#extending-the-analyzer-for-additional-pii-entities)
+- [Extending to additional languages](#extending-to-additional-languages)
+
+## Introduction to recognizer development
+
+Entity recognizers are Python objects capable of detecting one or more entities in a specific language.
+In order to extend Presidio's detection capabilities to new types of PII entities,
+these `EntityRecognizer` objects should be added to the existing list of recognizers.
+
+## Types of recognizer classes in Presidio
+
+The following class diagram shows the different types of recognizer families Presidio contains.
+
+![Recognizers class diagram](../assets/recognizers_class_diagram.png)
+
+- The `EntityRecognizer` is an abstract class for all recognizers.
+- The `RemoteRecognizer` is an abstract class for calling external PII detectors.
+See more info [here](#creating-a-remote-recognizer).
+- The `LocalRecognizer` is an abstract class for all recognizers running within the Presidio-analyzer process.
+- The `PatternRecognizer` is an class for supporting regex and deny-list based recognition logic,
```

Comment:
perhaps (just a suggestion):
- Abstract class `LocalRecognizer` - implemented by all recognizers running within the Presidio-analyzer process. 
- Class `PatternRecognizer` -  extended by regex and deny-list based recognition logic. It includes validation (e.g., with checksum) and context support. See an example [here](#simple-example).

Reply:
I find the sentence "extended by regex and deny-list based recognition logic." a bit confusing. The `PatternRecognizer` is the class which enables the logic of regex and deny-lists, it is not extended by it. If the current isn't clear enough, do you have a suggestion for rephrasing?

Reply:
Maybe contains regex and deny-list based.. ?

Reply:
This is still confusing. The pattern recognizer class doesn't contain the regex logic (which is what a specific PII recognizer would contain). It contains helper function for users to implement regex.